src/tests/ci.md

```diff
@@ -66,19 +109,169 @@ pr:
+override the cache for the others. Instead, we store the images under different
+tags, identifying them with a custom hash made from the contents of all the
+Dockerfiles and related scripts.
+
+[ghcr.io]: https://ghcr.io/rust-lang-ci/rust-ci
+[Docker registry caching]: https://docs.docker.com/build/cache/backends/registry/
+
+### LLVM caching with sccache
+
+We build some C/C++ stuff in various CI jobs, and we rely on [sccache] to cache
+the intermediate LLVM artifacts. Sccache is a distributed ccache developed by
+Mozilla, which can use an object storage bucket as the storage backend. In our case,
+the artefacts are uploaded to an S3 bucket that we control (`rust-lang-ci-sccache2`).
+
+[sccache]: https://github.com/mozilla/sccache
+
+## Custom tooling around CI
+
+During the years we developed some custom tooling to improve our CI experience.
+
+### Rust Log Analyzer to show the error message in PRs
+
+The build logs for `rust-lang/rust` are huge, and it’s not practical to find
+what caused the build to fail by looking at the logs. To improve the
+developers’ experience we developed a bot called [Rust Log Analyzer][rla] (RLA)
+that receives the build logs on failure and extracts the error message
+automatically, posting it on the PR.
+
+The bot is not hardcoded to look for error strings, but was trained with a
+bunch of build failures to recognize which lines are common between builds and
+which are not. While the generated snippets can be weird sometimes, the bot is
+pretty good at identifying the relevant lines even if it’s an error we've never
+seen before.
+
+[rla]: https://github.com/rust-lang/rust-log-analyzer
+
+### Toolstate to support allowed failures
+
+The `rust-lang/rust` repo doesn’t only test the compiler on its CI, but also a
+variety of tools and documentation. Some documentation is pulled in via git
+submodules. If we blocked merging rustc PRs on the documentation being fixed,
+we would be stuck in a chicken-and-egg problem, because the documentation's CI
+would not pass since updating it would need the not-yet-merged version of
+rustc to test against (and we usually require CI to be passing).
+
+To avoid the problem, submodules are allowed to fail, and their status is
+recorded in [rust-toolstate]. When a submodule breaks, a bot automatically
+pings the maintainers so they know about the breakage, and it records the
+failure on the toolstate repository. The release process will then ignore
+broken tools on nightly, removing them from the shipped nightlies.
+
+While tool failures are allowed most of the time, they’re automatically
+forbidden a week before a release: we don’t care if tools are broken on nightly
+but they must work on beta and stable, so they also need to work on nightly a
+few days before we promote nightly to beta.
+
+More information is available in the [toolstate documentation].
+
+[rust-toolstate]: https://rust-lang-nursery.github.io/rust-toolstate
+[toolstate documentation]: ../toolstate.md
```

Comment:
```
    ┌─ tests/ci.md:264:38
    │
264 │ More information is available in the [toolstate documentation].
    │                                      ^^^^^^^^^^^^^^^^^^^^^^^^^ File not found: ../toolstate.md
```

Can you please fix it @Kobzol?

Reply:
Thanks! https://github.com/rust-lang/rustc-dev-guide/pull/2021